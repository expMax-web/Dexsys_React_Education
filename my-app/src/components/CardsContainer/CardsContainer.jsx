import React from "react";

import {CharacterCard} from "../CharacterCard/CharacterCard";
import {UpButton} from "../UpButton/UpButton";

import styles from "./CardsContainer.module.scss";

export const CardsContainer = ({data}) => {
  return (
    <main className={styles.Container}>
    {data?.characters?.results.map((character) => (
      <CharacterCard image={character.image} name={character.name} />
    ))}
    <UpButton>Вверх</UpButton>
  </main>
  );
};
