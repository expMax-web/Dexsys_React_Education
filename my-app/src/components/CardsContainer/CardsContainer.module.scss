.Container {
  padding: 0px 200px 0px 200px;
  width: 100%;
  height: 100%;
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: space-between;
  background-color: #f4f7fc;
  &_dark {
    padding: 0px 200px 0px 200px;
    width: 100%;
    height: 100%;
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    justify-content: space-between;
    background-color: #636363;
  }
}
