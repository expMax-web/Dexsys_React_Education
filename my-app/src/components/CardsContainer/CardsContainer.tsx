import React from "react";
import { useState } from "react";
import cn from "classnames";

import { CharacterCard } from "../CharacterCard/CharacterCard";
import { UpButton } from "../UpButton/UpButton";
import { Maybe, Character } from "../../api/types";
import { useTheme } from "../../hooks/useTheme";

import styles from "./CardsContainer.module.scss";

interface CardsContainerProps {
  characters: Maybe<Character[]>;
}

export const CardsContainer: React.FC<CardsContainerProps> = ({
  characters,
}) => {
  const [showScroll, setShowScroll] = useState<Boolean>(false);

  const checkScrollTop = () => {
    if (!showScroll && window.pageYOffset > 300) {
      setShowScroll(true);
    }
    if (showScroll && window.pageYOffset <= 300) {
      setShowScroll(false);
    }
  };

  window.addEventListener("scroll", checkScrollTop);

  const scrollToTop = () => {
    window.scrollTo({
      top: 0,
      behavior: "smooth",
    });
    setShowScroll(false);
  };
  const { isDark } = useTheme();
  return (
    <main
      className={cn(styles.Container, {
        [styles.Container_Dark]: isDark,
      })}
    >
      {characters &&
        characters
          ?.filter((character) => character)
          .map((character: Character) => (
            <CharacterCard
              image={character.image}
              name={character.name}
              key={character.id}
              id={character.id}
            />
          ))}
      {showScroll && <UpButton scrollToTop={scrollToTop}>â–²</UpButton>}
    </main>
  );
};
