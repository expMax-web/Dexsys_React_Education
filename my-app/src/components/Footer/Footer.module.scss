.Footer {
  background-color: #86bcfc;
  padding-top: 15px;
  height: 500px;
  &_Dark {
    background-color: #23262b;
  }
}

.FooterTitle {
  font-weight: 500;
  &_Dark {
    color: white;
  }
}
