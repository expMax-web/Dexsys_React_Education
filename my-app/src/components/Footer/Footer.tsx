import React from "react";
import cn from "classnames";

import { useDarkTheme } from "../../hooks/useDarkTheme";
import { FeedBackForm } from "../FeedBackForm/FeedBackForm";

import styles from "./Footer.module.scss";

export const Footer: React.FC = () => {
  const { isDark } = useDarkTheme();
  return (
    <footer
      className={cn(styles.Footer, {
        [styles.Footer_Dark]: isDark,
      })}
    >
      <FeedBackForm />
    </footer>
  );
};
