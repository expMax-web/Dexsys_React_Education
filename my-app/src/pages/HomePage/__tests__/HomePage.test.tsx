import React from "react";
import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";

import { CardsContainer } from "../../../components/CardsContainer/CardsContainer";
import { Footer } from "../../../components/Footer/Footer";

const characters: any = [
  {
    id: "1",
    name: "Rick Sanchez",
    image: "https://rickandmortyapi.com/api/character/avatar/1.jpeg",
  },
  {
    id: "2",
    name: "Morty Smith",
    image: "https://rickandmortyapi.com/api/character/avatar/2.jpeg",
  },
  {
    id: "3",
    name: "Summer Smith",
    image: "https://rickandmortyapi.com/api/character/avatar/3.jpeg",
  },
];

describe("HomePage test suite", () => {
  test("should be render correctly ", () => {
    const { container } = render(
      <BrowserRouter>
        <div>
          <CardsContainer characters={characters} />
          <Footer />
        </div>
      </BrowserRouter>
    );
  });
});
