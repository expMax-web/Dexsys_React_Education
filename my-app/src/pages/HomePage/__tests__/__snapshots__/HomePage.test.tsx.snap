// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage test suite should be render correctly  1`] = `
<div>
  <div
    class="Container"
  >
    <span
      aria-busy="true"
      aria-live="polite"
    >
      <span
        class="react-loading-skeleton SkeletonCharacterCard"
      >
        ‌
      </span>
      <br />
    </span>
    <span
      aria-busy="true"
      aria-live="polite"
    >
      <span
        class="react-loading-skeleton SkeletonCharacterCard"
      >
        ‌
      </span>
      <br />
    </span>
    <span
      aria-busy="true"
      aria-live="polite"
    >
      <span
        class="react-loading-skeleton SkeletonCharacterCard"
      >
        ‌
      </span>
      <br />
    </span>
    <span
      aria-busy="true"
      aria-live="polite"
    >
      <span
        class="react-loading-skeleton SkeletonCharacterCard"
      >
        ‌
      </span>
      <br />
    </span>
    <span
      aria-busy="true"
      aria-live="polite"
    >
      <span
        class="react-loading-skeleton SkeletonCharacterCard"
      >
        ‌
      </span>
      <br />
    </span>
    <span
      aria-busy="true"
      aria-live="polite"
    >
      <span
        class="react-loading-skeleton SkeletonCharacterCard"
      >
        ‌
      </span>
      <br />
    </span>
    <span
      aria-busy="true"
      aria-live="polite"
    >
      <span
        class="react-loading-skeleton SkeletonCharacterCard"
      >
        ‌
      </span>
      <br />
    </span>
    <span
      aria-busy="true"
      aria-live="polite"
    >
      <span
        class="react-loading-skeleton SkeletonCharacterCard"
      >
        ‌
      </span>
      <br />
    </span>
  </div>
</div>
`;
